snorkel/mtl/model.py

```diff
@@ -0,0 +1,402 @@
+
+        if self.config["device"] >= 0:
+            if torch.cuda.is_available():
+                logging.info(f"Moving model to GPU " f"(cuda:{self.config['device']}).")
+                self.to(torch.device(f"cuda:{self.config['device']}"))
+            else:
+                logging.info("No cuda device available. Switch to cpu instead.")
+
+    def _build_network(self, tasks):
+        """Build the MTL network using all tasks"""
+
+        if not isinstance(tasks, Iterable):
+            tasks = [tasks]
+        for task in tasks:
+            if task.name in self.task_names:
+                raise ValueError(
+                    f"Found duplicate task {task.name}, different task should use "
+                    f"different task name."
+                )
+            if not isinstance(task, Task):
+                raise ValueError(f"Unrecognized task type {task}.")
+            self.add_task(task)
+
+    def add_task(self, task):
+        """Add a single task into MTL network"""
+
+        # Combine module_pool from all tasks
+        for key in task.module_pool.keys():
+            if key in self.module_pool.keys():
+                if self.config["dataparallel"]:
+                    task.module_pool[key] = nn.DataParallel(self.module_pool[key])
+                else:
+                    task.module_pool[key] = self.module_pool[key]
+            else:
+                if self.config["dataparallel"]:
+                    self.module_pool[key] = nn.DataParallel(task.module_pool[key])
+                else:
+                    self.module_pool[key] = task.module_pool[key]
+        # Collect task names
+        self.task_names.add(task.name)
+        # Collect task flows
+        self.task_flows[task.name] = task.task_flow
+        # Collect loss functions
+        self.loss_funcs[task.name] = task.loss_func
+        # Collect output functions
+        self.output_funcs[task.name] = task.output_func
+        # Collect scorers
+        self.scorers[task.name] = task.scorer
+
+        # Move model to specified device
+        self._move_to_device()
+
+    def update_task(self, task):
+        """Update a existing task in MTL network"""
+
+        # Update module_pool with task
+        for key in task.module_pool.keys():
+            # Update the model's module with the task's module
+            if self.config["dataparallel"]:
+                self.module_pool[key] = nn.DataParallel(task.module_pool[key])
+            else:
+                self.module_pool[key] = task.module_pool[key]
+        # Update task flows
+        self.task_flows[task.name] = task.task_flow
+        # Update loss functions
+        self.loss_funcs[task.name] = task.loss_func
+        # Update output functions
+        self.output_funcs[task.name] = task.output_func
+        # Collect scorers
+        self.scorers[task.name] = task.scorer
+
+        # Move model to specified device
+        self._move_to_device()
+
+    def remove_task(self, task_name):
```

Comment:
How often is it necessary to update or remove tasks...?  If answer is close to never, I would remove these pieces of code